src/goto-symex/symex_assign.h

```diff
@@ -20,6 +20,49 @@ class byte_extract_exprt;
 class ssa_exprt;
 struct symex_configt;
 
+/// Expression in which some part is missing and can be substituted for another
+/// expression.
+///
+/// For instance, a skeleton `☐[index]` where `☐` is the missing part, can be
+/// applied to an expression `x` to get `x[index]` (see
+/// \ref expr_skeletont::apply). It can also be composed with another skeleton,
+/// let say `☐.some_field` which would give the skeleton `☐.some_field[index]`
+/// (see \ref expr_skeletont::compose).
+class expr_skeletont final
```

Comment:
:bulb: I wouldn't have guessed the behaviour of this expression from the name. Perhaps `algebraic_expr`, `partial_expr` `expr_template`?

Reply:
Yes I couldn't find a very good name, but `algebraic_expr` could be mistaken for a special type of `exprt` (like expr containing symbols), `expr_template` would be expected to be a template, `partial_expr` is as good as `expr_skeleton` I think.